src/parser.js

```diff
@@ -530,41 +538,27 @@ function mapIfStatement(node, meta) {
   );
 }
 
-function isStatement(expr) {
-  const type = expr.constructor.name;
-  switch (type) {
-    case 'Literal':
-      if (expr.value === 'break' || expr.value === 'continue') {
-        return true;
-      }
-      return false;
-    case 'Throw':
-    case 'For':
-    case 'While':
-    case 'Return':
-    case 'If':
-    case 'Break':
-      return true;
-    default:
-      return false;
-  }
-}
+function isTernaryOperation(node, isExpressionCondition) {
```

Comment:
Yeah, my vote would be to do this as a transform instead of passing the state around.